classes/statistics/PKPTemporaryTotalsDAO.php

```diff
@@ -107,8 +108,8 @@ public function removeDoubleClicks(int $counterDoubleClickTimeFilter): void
      */
     public function compileContextMetrics(string $loadId): void
     {
-        $date = substr($loadId, -12, 8);
-        DB::table('metrics_context')->where('load_id', '=', $loadId)->orWhere('date', '=', DB::raw("DATE({$date})"))->delete();
+        $date = DateTimeImmutable::createFromFormat('Ymd', substr($loadId, -12, 8));
+        DB::table('metrics_context')->where('load_id', '=', $loadId)->orWhereDate('date', '=', $date->format('Y-m-d'))->delete();
```

Comment:
I think `orWhereDate` will take a `DateTimeInterface` for `$value` directly (and `DateTimeImmutable::createFromFormat` returns one -- so there shouldn't be a need to convert it back to a string with `$date->format(...)`. (Also below.)

Reply:
Oh, yes, good catch!
Changed...